### PR TITLE
Adding cone primitives.

### PR DESCRIPTION
**Pull request by bperseghetti**
# 🦟 Bug fix

## Summary
This helps add the missing cone geometry for primitive/basic parametric shapes:

conetopple
cone

And is also valuable for visualizations of emitters/source that typically have conic-based spread as seen in this acoustic attack on an IMU by showing the affected area:

drone_attack

Associated PRs:
- https://github.com/gazebosim/gz-gui/pull/621
- https://github.com/gazebosim/gz-math/pull/594
- https://github.com/gazebosim/gz-msgs/pull/442
- https://github.com/gazebosim/gz-physics/pull/639
- https://github.com/gazebosim/gz-rendering/pull/1003
- https://github.com/gazebosim/gz-sim/pull/2410
- https://github.com/gazebosim/sdformat/pull/1418

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [x] Updated documentation (as needed)
- [x] Updated migration guide (as needed)
- [x] Consider updating Python bindings (if the library has them)
- [x] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [x] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: include/gz/rendering/ParticleEmitter.hh [hunk 1]
```diff
@@ -47,8 +47,11 @@ namespace gz
       /// \brief Ellipsoid emitter.
       EM_ELLIPSOID        = 3,
 
+      /// \brief Cone emitter.
+      EM_CONE             = 4,
```
Comment (iche033) [inline include/gz/rendering/ParticleEmitter.hh:51]: ogre 1.x and ogre 2.x do not support cone [particle emitters](https://ogrecave.github.io/ogre/api/1.12/_particle-_scripts.html#Standard-Particle-Emitters). So let's not add it to the enums yet until we have an engine that supports it.

File: include/gz/rendering/ParticleEmitter.hh [hunk 2]
```diff
@@ -86,8 +89,10 @@ namespace gz
       /// the emmiter type:
       ///   - EM_POINT: The area is ignored.
       ///   - EM_BOX: The area is interpreted as width X height X depth.
+      ///   - EM_CONE: The area is interpreted as the bounding box of the
```
Comment (iche033) [inline include/gz/rendering/ParticleEmitter.hh:92]: we can remove this

File: ogre2/src/Ogre2ParticleEmitter.cc [hunk 3]
```diff
@@ -75,6 +75,7 @@ static const std::array<std::string, EmitterType::EM_NUM_EMITTERS>
       {
         "Point",
         "Box",
+        "Cone",
```
Comment (iche033) [inline ogre2/src/Ogre2ParticleEmitter.cc:78]: we can remove this since it's not supported in ogre

File: ogre2/src/Ogre2ParticleEmitter.cc [hunk 4]
```diff
@@ -154,6 +155,7 @@ void Ogre2ParticleEmitter::SetEmitterSize(const gz::math::Vector3d &_size)
       break;
     }
     case EmitterType::EM_BOX:
+    case EmitterType::EM_CONE:
```
Comment (iche033) [inline ogre2/src/Ogre2ParticleEmitter.cc:158]: we can remove this since it's not supported yet

**Review by iche033 — COMMENTED** [review]
we can add cone shape to markers but let's not add it to particle emitters yet since we don't have an engine that supports it

File: include/gz/rendering/ParticleEmitter.hh [hunk 5]
```diff
@@ -86,8 +86,10 @@ namespace gz
...
```
Comment (iche033) [inline include/gz/rendering/ParticleEmitter.hh:89]: can you remove this one as well?

Comment (bperseghetti) [reply include/gz/rendering/ParticleEmitter.hh]: @iche033 done, sorry I missed that one.

**Review by iche033 — APPROVED** [review]
lgtm!

**Pull request merged**